Library/Formula/ocilib.rb

```diff
@@ -0,0 +1,47 @@
+class OracleHomeVar < Requirement
+  fatal true
+  satisfy ENV["ORACLE_HOME"]
+
+  def message; <<-EOS.undent
+      To use this formula you have to set the ORACLE_HOME environment variable.
+      You might need to create a symbolic link without version number
+      for the libclntsh library.
+      Check Oracle Instant Client documentation for more information.
+    EOS
+  end
+end
+
+class Ocilib < Formula
+  desc "Open source C and C++ library for accessing Oracle Databases"
+  homepage "http://www.ocilib.net/"
+  url "https://github.com/vrogier/ocilib/releases/download/v4.1.0/ocilib-4.1.0-gnu.tar.gz"
+  sha256 "d86c1e6d5ef33541299996b76497a1271498cdcf5720ee0f1248a0b78dfb3a0c"
+
+  head "https://github.com/vrogier/ocilib.git"
+
+  option "with-charset-wide", "Use wide charset with Oracle OCI libraries"
+
+  depends_on OracleHomeVar
```

Comment:
I've once developed a DB driver [component](https://github.com/vszakats/harbour-core/tree/master/contrib/sddoci) that depends on `ocilib` as part of [Harbour](https://github.com/vszakats/harbour-core) part out of fun and because I used Oracle DB while attending lectures. Since I still use Harbour (though not this specific part of it), I try to keep it maintained, even those parts I don't currently need. Having brew formulae for its dependencies do help to keep it maintained on OS X.